### PR TITLE
feature: Add systemd unit for linux systems

### PR DESCRIPTION
**Pull request by jcpunk**
This should close out https://github.com/intel/pcm/issues/391

### CODE REVIEW

**Review by rdementi — CHANGES_REQUESTED** [review]
thanks for submitting the patch. Please address the issues reported by the checks.

**Comment by jcpunk** [comment]
I think I've addressed the issues, but I'm not really sure.....

**Comment by jcpunk** [comment]
I'm not thinking the MacOS build error is from my patch.

**Review by rdementi — CHANGES_REQUESTED** [review]
thanks for the attempt to fix the issues. The remaining failure is unrelated. However having a requirement to provide -DLINUX_SYSTEMD_UNITDIR for all kinds of builds is not a standard practice. Can we avoid this and/or make the feature optional?

**Comment by jcpunk** [comment]
> Can we avoid this and/or make the feature optional?

systemd itself defines the expected unit dir as /usr/lib/systemd/system.  Any place the build is running `make install` will need to have some sort of alternate path provided so the CI tests don't try to write outside their permissions.

I'm absolutely open to ideas how to do this better, but a bit out of my depth in CMake.

**Comment by rdementi** [comment]
I have added @markovamaria maybe she has an idea

**Comment by rdementi** [comment]
yes. An idea would be to have this enabled only if the project was configured with a new systemd-unit option

cmake .. <enable-systemd-unit-option>
make install

otherwise the old paths should be used

**Comment by jcpunk** [comment]
Ok, I think I've taken a stab at that.

File: src/pcm-sensor-server.service.in [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[Unit]
+Description=Intel Process Counter Monitor (PCM) Sensor Service
```
Comment (rdementi) [inline src/pcm-sensor-server.service.in:2]: Please change: Process -> Performance

Comment (jcpunk) [reply src/pcm-sensor-server.service.in]: Oops, fixed.

File: .github/workflows/ci-cmake-options.yml [hunk 2]
```diff
@@ -62,6 +62,15 @@ jobs:
...
 
+    - name: CMake Include systemd unit
+      run: |
+        cmake --version
+        rm -rf ${{ github.workspace }}/build
+        cmake -B ${{ github.workspace }}/build -DCMAKE_INSTALL_PREFIX=${{ github.workspace }}/build -DLINUX_SYSTEMD=TRUE -DLINUX_SYSTEMD_UNITDIR=${{ github.workspace }}/build
+        cd ${{ github.workspace }}/build
+        export VERBOSE=1
+        make install -j$(nproc)
+
```
Comment (rdementi) [inline .github/workflows/ci-cmake-options.yml:73]: this workflow does not run on this repo. Could you please extend .github/workflows/linux_make.yml as well? .github/workflows/linux_make.yml will run here

**Review by rdementi — CHANGES_REQUESTED** [review]
thanks a lot for the updated change. There is just one request to add an additional github workflow

File: .github/workflows/ci-cmake-options.yml (hunk 2 above)
Comment (jcpunk) [reply .github/workflows/ci-cmake-options.yml:73]: Can do, does the latest push look better?

Comment (rdementi) [reply .github/workflows/ci-cmake-options.yml:73]: adding a new "build-systemd" job would be cleaner

Comment (jcpunk) [reply .github/workflows/ci-cmake-options.yml:73]: Done.

**Comment by rdementi** [comment]
if it works, should the make step in https://github.com/intel/pcm/actions/runs/4066019364/jobs/7001721969 print something like "-- Installing: ... pcm-sensor-server.service"

**Comment by jcpunk** [comment]
I wonder if the print is being suppressed since I'm installing to the same location as where I build it.  I'll try moving to another location for the "fake install"

**Review by opcm — CHANGES_REQUESTED** [review]
thanks a lot for your contribution and addressing change requests in a timely manner

**Review by rdementi — APPROVED** [review]
thanks a lot for your contribution and addressing change requests in a timely manner

**Pull request merged**